tortoise/backends/asyncpg/client.py

```diff
@@ -42,85 +42,61 @@ def __init__(self, user: str, password: str, database: str, host: str, port: Sup
         self.host = host
         self.port = int(port)  # make sure port is int type
 
-        self.dsn = self.DSN_TEMPLATE.format(
-            user=self.user,
-            password=self.password,
-            host=self.host,
-            port=self.port,
-            database=self.database
-        )
-        self._db_pool = None  # Type: Optional[asyncpg.pool.Pool]
+        # self._db_pool = None  # Type: Optional[asyncpg.pool.Pool]
         self._connection = None  # Type: Optional[asyncpg.Connection]
 
         self._transaction_class = type(
             'TransactionWrapper', (TransactionWrapper, self.__class__), {}
         )
 
-    async def create_connection(self) -> None:
+    async def create_connection(self, with_db: bool) -> None:
+        dsn = self.DSN_TEMPLATE.format(
+            user=self.user,
+            password=self.password,
+            host=self.host,
+            port=self.port,
+            database=self.database if with_db else ''
+        )
         try:
-            if not self.single_connection:
-                self._db_pool = await asyncpg.create_pool(self.dsn)
-            else:
-                self._connection = await asyncpg.connect(self.dsn)
+            self._connection = await asyncpg.connect(dsn)
             self.log.debug(
-                'Created connection with params: user=%s database=%s host=%s port=%s',
-                self.user, self.database, self.host, self.port
+                'Created connection %s with params: user=%s database=%s host=%s port=%s',
+                self._connection, self.user, self.database, self.host, self.port
```

Comment:
May be we should change it to `.format` here for uniformity of string formatting?

Reply:
That is a logging message. For performance reasons we should use %s and commas. It then allows the log to become a no-op when you don't have debug level logging configured.
Using string interpolation in the logging functions is considered a code smell.

Reply:
Ah, yeah, wrong on my part 👍